src/imp/to_fun.rs

```diff
@@ -83,7 +99,7 @@ fn wrap(nxt: Option<fun::Pattern>, term: fun::Term, ask: bool) -> StmtToFun {
 }
 
 impl Stmt {
-  fn into_fun(self) -> Result<StmtToFun, String> {
+  fn into_fun(self) -> Result<StmtToFun, Diagnostics> {
```

Comment:
What is the point of returning a new diagnotics context if we only ever return a single string?

Reply:
I changed this one because `into_fun` calls Definition's `to_fun` in this section:
```rs
Stmt::LocalDef { def, nxt } => {
  let (ask, nxt_pat, nxt) = take(*nxt)?;
  let def = def.to_fun()?; // <----- here
  let term = fun::Term::Def { def, nxt: Box::new(nxt) };
  wrap(nxt_pat, term, ask)
}
```
and I had changed that function's signature to return `Diagnostics` so it could report its origin. To keep returning strings in this function we could create a function to extract error message from `Diagnostics` with a single `Diagnostic`.
What do you think?

Reply:
But if we're returning only strings, shouldn't the `From<String>` implementation of `Diagnostics` take care of converting it at the place you point out? (`?` calls something like `into_residual` which calls a `into`)

In the same way that the `?` inside `into_fun` is converting a string into a diagnostic, wouldn't it already do the same here? I don't understand why changing the signature is necessary.

Reply:
You're right; originally I'd changed `into_fun` to return Diagnostics because of a problem I was having with multiple `Error:` messages being printed into the error string and with what you said I noticed I could do it in another way; I changed it back into returning strings.